plasmapy/particles/ionization_state.py

```diff
@@ -113,7 +261,20 @@ def __init__(
     ):
         """Initialize an `~plasmapy.particles.IonizationState` instance."""
 
-        self._particle_instance = particle
+        if particle.is_ion or particle.is_category(require=("uncharged", "element")):
+            if ionic_fractions is None:
+                ionic_fractions = np.zeros(particle.atomic_number + 1)
+                ionic_fractions[particle.integer_charge] = 1.0
+                particle = Particle(
+                    particle.isotope if particle.isotope else particle.element
+                )
+            else:
+                raise ParticleError(
```

Comment:
Good idea!

Reply:
I created these exceptions, but applying them would increase the scope of this PR more than I'd like since it would involve digging into `Particle` and its methods in more detail.  

- [ ] When this is merged, create an issue on implementing `ParticleTypeError` and `ParticleValueError` in `plasmapy.particles`